mypy/types.py

```diff
@@ -51,7 +51,9 @@ def serialize(self) -> JsonDict:
         raise NotImplementedError('Cannot serialize {} instance'.format(self.__class__.__name__))
 
     @classmethod
-    def deserialize(cls, data: JsonDict) -> 'Type':
+    def deserialize(cls, data: Union[JsonDict, str]) -> 'Type':
+        if isinstance(data, str):
+            return Instance.deserialize(data)
```

Comment:
Actually you can just inline the code you added to `Instance.deserialize()` here -- then you won't have to change the argument to the latter to a Union.